test/test_external.py

```diff
@@ -331,5 +339,61 @@ def test_can_load_tabular_model_with_different_widget_data(hypothetical_readme):
         check_dataset(io.config, hypothetical_readme)
 
 
+@pytest.mark.parametrize(
+    "config, answer",
+    [
+        ({"version": "3.3", "enable_queue": True}, True),
+        ({"version": "3.3", "enable_queue": False}, False),
+        ({"version": "3.2", "enable_queue": False}, False),
+        ({"version": "3.2", "enable_queue": True}, True),
+        ({"version": "3.1.3", "enable_queue": True}, False),
+        ({"version": "3.1.3", "enable_queue": False}, False),
+    ],
+)
+def test_use_websocket_after_315(config, answer):
+    assert use_websocket(config) == answer
+
+
+class AsyncMock(MagicMock):
+    async def __call__(self, *args, **kwargs):
+        return super(AsyncMock, self).__call__(*args, **kwargs)
+
+
+@pytest.mark.asyncio
+async def test_get_pred_from_ws():
+    mock_ws = AsyncMock(name="ws")
+    messages = [
+        json.dumps({"msg": "estimation"}),
+        json.dumps({"msg": "send_data"}),
+        json.dumps({"msg": "process_generating"}),
+        json.dumps({"msg": "process_completed", "output": {"data": ["result!"]}}),
+    ]
+    mock_ws.recv.side_effect = messages
+    data = json.dumps({"data": ["foo"], "fn_index": "foo"})
+    output = await get_pred_from_ws(mock_ws, data)
+    assert output == {"data": ["result!"]}
+    mock_ws.send.assert_called_once_with(data)
+
+
+@pytest.mark.asyncio
+async def test_get_pred_from_ws_raises_if_queue_full():
+    mock_ws = AsyncMock(name="ws")
+    messages = [json.dumps({"msg": "queue_full"})]
+    mock_ws.recv.side_effect = messages
+    data = json.dumps({"data": ["foo"], "fn_index": "foo"})
+    with pytest.raises(gradio.Error, match="Queue is full!"):
+        await get_pred_from_ws(mock_ws, data)
+
+
+@pytest.mark.skipif(sys.version_info < (3, 8), reason="Async mocks don't work for 3.7")
+def test_respect_queue_when_load_from_config():
+    with unittest.mock.patch("websockets.connect"):
+        with unittest.mock.patch(
+            "gradio.external.get_pred_from_ws", return_value={"data": ["foo"]}
+        ):
+            interface = gr.Interface.load("spaces/multimodalart/saymyname")
```

Comment:
I think you should use your own Space in the test in case `multimodalart` decides to change theirs. This test should also be marked as `flaky`.